src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -355,10 +355,16 @@ emit_simd_ins_for_binary_op (MonoCompile *cfg, MonoClass *klass, MonoMethodSigna
 				instc0 = OP_FDIV;
 				break;
 			}
+#ifdef TARGET_ARM64
```

Comment:
Why are we changing this behavior?

Reply:
`Max` was non-determinstic across platforms, which is undesirable. It's instead being split into `MaxNative` which does the platform specific thing and `Max` which does the deterministic thing.

This is similar to the same decision that was made for floating-point to integer conversions and why we have `ConvertToInt32` and `ConvertToInt32Native` now.